nullaway/src/main/java/com/uber/nullaway/dataflow/AccessPath.java

```diff
@@ -255,6 +255,34 @@ public static AccessPath getAccessPathForNodeWithMapGet(Node node, @Nullable Typ
     }
   }
 
+  /**
+   * Gets corresponding AccessPath for a class field element.
+   *
+   * @param receivers The list of receivers.
+   * @param field The class field element.
+   * @return corresponding AccessPath.
+   */
+  public static AccessPath fromFieldAndBase(List<Element> receivers, Element field) {
```

Comment:
Another thought here.  I wonder if the code would overall be simpler if we instead had an API `fromMemberSelectTree(MemberSelectTree tree)`, and possibly overloads for other types of `Tree`s we need to handle.  Then all the logic about extracting `Element`s into a `List<AccessPathElement>` would be internal to the `AccessPath` class, which matches the current structure of the code, and clients would be much simpler.  What do you think?

Reply:
I added the following methods to make it more simple. All the methods below have ```@Nonnull``` parameters and as you mentioned, all the logic for creating ```List<AccessPathElement>``` is now internal to the ```AcessPath``` class.

```java
/**
   * Constructs an access path ending with the element in the argument. The receiver is the method
   * receiver itself.
   *
   * @param element The receiver element.
   * @return An access path ending with the element in the argument.
   */
  public static AccessPath fromElement(Element element)

  /**
   * Extends an access path with a class field element.
   *
   * @param receiverNode The receiver node.
   * @param field element of the class field.
   */
  public static AccessPath extendReceiverAccessPathWithField(Node receiverNode, Element field)

  /**
   * Extends an access path with a class field element.
   *
   * @param receiverTree The receiver tree.
   * @param field element of the class field
   */
  public static AccessPath extendReceiverAccessPathWithField(Tree receiverTree, Element field)
```